src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13775,6 +13573,31 @@ private void Release(ref ResetPoint state)
             base.Release(ref state.BaseResetPoint);
         }
 
+        private ref struct DisposableResetPoint
```

Comment:
Why does this need to be a `ref struct`? It doesn't hold any `ref`s.

Reply:
It keeps it on the stack, effectively forcing a single pattern for using it.